src/QuLDE.jl

```diff
@@ -126,23 +124,27 @@ struct QuLDEnonUnitParam{L, W, Q, NL, Nbit, VType <: GeneralMatrixBlock{1}, STyp
         WS1 = matblock(WS1)
         WS2 = matblock(WS2)
 
-        VT = rand(ComplexF64,2^l,2^l)
+        VT = rand(Complex{PType},2^l,2^l)
         VT[:,1] = 0.5*ones(2^l,1)
         VT = -1*qr(VT).Q
         WT = convert(typeof(VT),VT')
         VT = matblock(VT)
         WT = matblock(WT)
-        new{ComplexF64,L,k,l,nbit, typeof(V), typeof(VS1),typeof(VT)}(k,t,l,C_tilda,D_tilda,N,V,VS1,VS2,WS1,WS2,VT,WT,F)
+        new{PType, L, k, l, nbit, typeof(V), typeof(VS1),typeof(VT)}(k,t,l,C_tilda,D_tilda,N,V,VS1,VS2,WS1,WS2,VT,WT,F)
     end
 end
 
+getparam(blk::QuLDEUnitParam{PType}) where {PType} = PType
```

Comment:
if it returns the type, it would be better to use more precise name, e.g `get_param_type`